### PR TITLE
[1.20.x] Fix Reach Issues (and eclipse compilation issues)

### LINKED ISSUE #243: The player's attack range is increased with NeoForge
**Minecraft Version:** 1.20.2

**NeoForge Version:** NeoForge-20.2.26-beta

NeoForge:

https://github.com/neoforged/NeoForge/assets/68275908/cb7a0e27-c563-4700-8085-fd9837c334f0

Vanilla:

https://github.com/neoforged/NeoForge/assets/68275908/73fbcddf-efa7-4664-b7e3-a8f2f2ca1365

World:
[Reach.zip](https://github.com/neoforged/NeoForge/files/13254081/Reach.zip)

### PR DESCRIPTION
**Pull request by Shadows-of-Fire**
This PR addresses #243 and other reach changes that should have been fixed in porting.  It also fixes issues with the eclipse compiler that were introduced by #73

### CODE REVIEW

**Review by Technici4n — DISMISSED** [review]
You should use a proper compiler instead of making changes to the NeoForge source code due to its limitations.

**Comment by Shadows-of-Fire** [comment]
Like it or not, we support eclipse, so we have to be compatible with the eclipse compiler.

**Comment by sciwhiz12** [comment]
At some point, we ought to add some automated check to compile with the Eclipse compiler. Though that may be somewhat difficult, given how hardcoded Gradle seems to be with using a `javac` executable for compilation.

File: src/main/java/net/neoforged/neoforge/common/extensions/IPlayerExtension.java [hunk 1]
```diff
@@ -20,13 +20,13 @@ private Player self() {
     }
 
     /**
-     * The entity reach is increased by 3 for creative players, unless it is currently zero, which disables attacks and entity interactions.
+     * The entity reach is increased by 2 for creative players, unless it is currently zero, which disables attacks and entity interactions.
```
Comment (Technici4n) [inline src/main/java/net/neoforged/neoforge/common/extensions/IPlayerExtension.java:23]: Can you update the comment on `ENTITY_REACH` in `NeoForgeMod` as well?

**Review by Technici4n — COMMENTED** [review]
OK this looks good. Just one thing.

**Review by Technici4n — APPROVED** [review]

**Pull request merged**